forecasttools/data.py

```diff
@@ -0,0 +1,478 @@
+        )
+    # check that the loaded CSV has the needed columns
+    df_scan = pl.scan_csv(nhsn_dataset_path)
+    df_cols = df_scan.columns
+    required_cols = ["state", "date", "hosp"]
+    if len(set(required_cols).intersection(set(df_cols))) != len(
+        required_cols
+    ):
+        raise ValueError(
+            f"NHSN dataset missing required columns: {set(required_cols) - set(required_cols).intersection(set(df_cols))}"
+        )
+    # load dataset
+    nhsn = pl.read_csv(nhsn_dataset_path)
+    # check if provided jurisdictions are in NHSN jurisdictions
+    nhsn_juris = list(nhsn["state"].unique().to_numpy())
+    assert len(set(juris_subset).intersection(set(nhsn_juris))) == len(
+        juris_subset
+    ), f"There are jurisdictions present that are not found in the dataset.\nEntered {juris_subset}, Available: {nhsn_juris}"
+    # clean data and organize data, cleaning null values
+    nhsn = nhsn.filter(pl.col("hosp").is_not_null())
+    nhsn = nhsn.filter(pl.col("state").is_in(juris_subset))
+    nhsn = nhsn.with_columns(
+        pl.col("hosp").cast(pl.Int64),
+        pl.col("date").str.strptime(pl.Date, "%Y/%m/%d"),
+    )
+    nhsn = nhsn.filter(
+        (
+            pl.col("date")
+            >= pl.lit(start_date).str.strptime(pl.Date, "%Y-%m-%d")
+        )
+        & (
+            pl.col("date")
+            <= pl.lit(end_date).str.strptime(pl.Date, "%Y-%m-%d")
+        )
+    )
+    nhsn = nhsn.group_by("state").agg([pl.col("hosp")])
+
+    # spline basis function
+    def spline_basis(X, degree=4, df=8):
+        basis = patsy.dmatrix(
+            "bs(x, df=df, degree=degree, include_intercept=True) - 1",
+            {"x": X, "df": df, "degree": degree},
+            return_type="matrix",
+        )
+        return np.array(basis)
+
+    # spline regression model(s)
+    def model(basis_matrix, y=None):
+        # priors
+        shift = npro.sample("shift", dist.Normal(0.0, 5.0))
+        beta_coeffs = npro.sample(
+            "beta_coeffs", dist.Normal(jnp.zeros(basis_matrix.shape[1]), 5.0)
+        )
+        shift_mu = jnp.dot(basis_matrix, beta_coeffs) + shift
+        mu_exp = jnp.exp(shift_mu)
+        alpha = npro.sample("alpha", dist.Exponential(1.0))
+        # likelihood
+        npro.sample("obs", dist.NegativeBinomial2(mu_exp, alpha), obs=y)
+
+    def model2(X, basis_matrix, y=None):
+        # priors
+        time_drift = npro.sample("time_drift", dist.Normal(0, 3.0))
+        drift_effect = time_drift * X
+        beta_coeffs = npro.sample(
+            "beta_coeffs", dist.Normal(jnp.zeros(basis_matrix.shape[1]), 3.0)
+        )
+        mu = jnp.dot(basis_matrix, beta_coeffs) + drift_effect
+        mu_exp = jnp.exp(mu)
+        alpha = npro.sample("alpha", dist.Exponential(1.0))
+        # likelihood
+        npro.sample("obs", dist.NegativeBinomial2(mu_exp, alpha), obs=y)
+
+    # define some shared inference values
+    random_seed = 2134312
+    num_samples, num_warmup = 1000, 500
```

Comment:
split into two lines

Reply:
Is this in a PEP, personal preference, readability suggestion, or something else?

Reply:
Readability. Seems like this makes it slightly harder to understand without any benefit?

Reply:
I will change this / avoid this pattern in the future w/ CDC but benefit in my mind is (1) shorter code and (2) communication that the parameters are very closely related.

Reply:
This example model run seems somewhat specific to have in the main package source code. Possibly better to have it generated in a tutorial/vignette and then loaded within the package itself from a human-readable saved format? @damonbayer any thoughts?

Reply:
`model2` is an artifact; I was curious about it's performance relative to `model` and wanted (want?) to have a second `idata` object for the same fitting period and jurisdictions for a future vignette(s) on model comparison (and evaluation). I feel more confident about one "simple" model being included in the package than two. However, I can see the case for removing the model from source once the example `idata` has been generated but including documentation on how the `idata` associated with `forecasttools` was generated.

Reply:
This was not specific to `model2`, but about both. My suggestion was more
> I can see the case for removing all models from source once the example idata has been generated but including documentation on how the idata associated with forecasttools was generated.

Reply:
Removing models from source sounds good; I will remove them.

Reply:
The models have been removed for the time being. I have yet to document them though.